Sources/HAP/Base/Accessory.swift

```diff
@@ -31,6 +35,37 @@ open class Accessory: JSONSerializable {
     public let info: Service.Info
     internal let services: [Service]
 
+    // An accessory implementation can set this flag to false if a device
+    // becomes unreachable for a period. If the accessory has a
+    // BridgingState Service, then the reachable property of that Service is set.
+    open var reachable = true {
+        didSet {
+            guard self.reachable == oldValue,
+                let characteristic = services
+                    .first(where: { $0.type == .bridgingState })?
+                    .characteristics
+                    .first(where: { $0.type == .reachable }) else {
+                        return
+            }
+            try? characteristic.setValue(self.reachable, fromConnection: nil)
+        }
+    }
+
+    // An accessory must provide a text identifier, which is guranteed to be
+    // unique. The implementation could provide the actual serial number of a
+    // device, or a MAC address or some other identifier which is not used on
+    // any other accessory.
+    //
+    // Device will check to ensure the serial numbers of all accessories added
+    // to a bridge are are unique.
+    //
+    // This is used for persistance of HomeKit AID's.
+    open var uniqueSerialNumber: String {
```

Comment:
We should call this `serialNumber`, dropping the `unique` prefix. Naming something a serial number is already implying uniqueness.

Reply:
Can do

Reply:
Please change this.